pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeUtils.java

```diff
@@ -89,31 +91,180 @@
+        }
+        if (instanceConfig.getFunctionDetails().getSource().getSubscriptionName() != null) {
+            functionConverGoConfig.setSubscriptionName(instanceConfig.getFunctionDetails().getSource().getSubscriptionName());
+        }
+
+        if (instanceConfig.getFunctionDetails().getSource().getInputSpecsMap() != null) {
+            for (String inputTopic : instanceConfig.getFunctionDetails().getSource().getInputSpecsMap().keySet()) {
+                functionConverGoConfig.setSourceSpecsTopic(inputTopic);
+            }
+        }
+
+        if (instanceConfig.getFunctionDetails().getSource().getTimeoutMs() != 0) {
+            functionConverGoConfig.setTimeoutMs(instanceConfig.getFunctionDetails().getSource().getTimeoutMs());
+        }
+
+        if (instanceConfig.getFunctionDetails().getSink().getTopic() != null) {
+            functionConverGoConfig.setSinkSpecsTopic(instanceConfig.getFunctionDetails().getSink().getTopic());
+        }
+
+        if (instanceConfig.getFunctionDetails().getResources().getCpu() != 0) {
+            functionConverGoConfig.setCpu(instanceConfig.getFunctionDetails().getResources().getCpu());
+        }
+
+        if (instanceConfig.getFunctionDetails().getResources().getRam() != 0) {
+            functionConverGoConfig.setRam(instanceConfig.getFunctionDetails().getResources().getRam());
+        }
+
+        if (instanceConfig.getFunctionDetails().getResources().getDisk() != 0) {
+            functionConverGoConfig.setDisk(instanceConfig.getFunctionDetails().getResources().getDisk());
+        }
+
+        if (instanceConfig.getFunctionDetails().getRetryDetails().getDeadLetterTopic() != null) {
+            functionConverGoConfig.setDeadLetterTopic(instanceConfig.getFunctionDetails().getRetryDetails().getDeadLetterTopic());
+        }
+
+        if (instanceConfig.getFunctionDetails().getRetryDetails().getMaxMessageRetries() != 0) {
+            functionConverGoConfig.setMaxMessageRetries(instanceConfig.getFunctionDetails().getRetryDetails().getMaxMessageRetries());
+        }
+
+        functionConverGoConfig.setKillAfterIdleMs(0);
+
+        DumperOptions dumperOptions = new DumperOptions();
+        dumperOptions.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        Yaml yaml = new Yaml(dumperOptions);
+        String output = yaml.dumpAs(functionConverGoConfig, Tag.MAP, null);
+        String pathName = "pulsar-functions/runtime/src/main/resources";
+        String fileName = String.format("%s_%s_%s.yml", functionConverGoConfig.getTenant(), functionConverGoConfig.getNameSpace(),
+                functionConverGoConfig.getName());
+        File ymlFile = new File(pathName + "/" + fileName);
+        try {
+            FileWriter fileWriter = new FileWriter(ymlFile);
+            fileWriter.write(output);
+            fileWriter.flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
it is a bad practice to use `e.printStackTrace()`. 

Also I don't think you should catch the exception here. You should let the function throw the exception and fail the operation. Because Go instance can not run correctly without the instance config.